ipam/provider/aws.go

```diff
@@ -0,0 +1,144 @@
+package provider
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	awsclient "github.com/giantswarm/e2eclients/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+// apiextensionsAWSConfigE2EChartValues is modified version of
+// e2etemplates.ApiextensionsAWSConfigE2EChartValues for IPAM test scenario.
+// Major difference is lack of subnet fields and need to replace clusterName
+// format string value. ClusterName is not based on env variable because there
+// would be too high risk for unintentional disruption of other tests if
+// ${CLUSTER_NAME} would be dynamically changed on the fly.
+const apiextensionsAWSConfigE2EChartValues = `commonDomain: ${COMMON_DOMAIN}
```

Comment:
Why do we have to have this here? I thought there is some place in which we hold all of them. Also the whole environment magic is again somewhere in the depths in the code and I am strictly against that. So when we get a chance to make this better somehow I would like to do so. For instance the provider should just be configured with the necessary bits and pieces.

Reply:
I copied this here because with current implementation this is specific to my test case. Refactoring existing code would be too much work in the scope of my story as I need to progress with that. I fully understand your concern with env variables so would it be ok if I implement the same workaround here as with `e2e-harness/framework.Guest` that there is config parameters and if those are missing, env variables are looked up?

Reply:
Yeah this would be a step into the right direction I think.